rs/sns/governance/canister/governance.did

```diff
@@ -21,6 +21,135 @@ type Action = variant {
   ExecuteGenericNervousSystemFunction : ExecuteGenericNervousSystemFunction;
   ManageLedgerParameters : ManageLedgerParameters;
   Motion : Motion;
+  RegisterExtension : RegisterExtension;
+  DeregisterExtension : DeregisterExtension;
+  ExecuteExtensionFunction : ExecuteExtensionFunction;
+};
+
+type ExistingCanister = record {
+  canister_id : opt principal;
+  wasm_module_hash : opt blob;
+};
+
+type NewCanister = record {
+  chunked_canister_wasm : opt ChunkedCanisterWasm;
+};
+
+type Installation = variant {
+  existing_canister : ExistingCanister;
+  new_canister : NewCanister;
+};
+
+type Icrc2Token = variant {
+  NativeSnsToken : record {};
+  Icp : record {};
+};
+
+type Icrc2Allowance = record {
+    token : opt Icrc2Token;
+    amount_e8s : opt nat64;
+};
+
+type Allowances = record {
+  icrc2_allowances : opt vec Icrc2Allowance;
+};
+
+// Used in responses, e.g., list_nervous_system_functions.
+type ExtensionType = variant {
+  TreasuryManager;
+};
+
+// Used in requests, e.g., manage_neuron (to submit a RegisterExtension proposal).
+type TreasuryManagerInit = record {
+  // The initial allowances for the Treasury Manager.
+  initial_allowances : opt Allowances;
+};
+
+type ExtensionInit = variant {
+  TreasuryManagerInit : TreasuryManagerInit;
+};
+
+// Native proposal type: RegisterExtension.
+type RegisterExtension = record {
+  // Determines the type of installation for the extension canister; e.g., whether it is
+  // a new canister to be installed by the SNS or an existing one.
+  installation : opt Installation;
+
+  extension_init : opt ExtensionInit;
+};
+
+// Native proposal type: DeregisterExtension.
+type DeregisterExtension = record {
+  extension_canister_id : opt principal;
+};
+
+type Deposit = record {
+  allowances : opt Allowances;
+};
+
+type Withdraw = record {};
+
+type TreasuryManagerFunctionArg = variant {
+  Deposit : Deposit;
+  Withdraw : Withdraw;
+};
+
+// Used in requests, e.g., manage_neuron (to submit an ExecuteExtensionFunction proposal).
+type ExtensionFunctionArg = variant {
```

Comment:
If the extensions are using the same custom proposals framework, I wonder if we need to fully specify all these things in the governance.did?

OTOH, it's nice to have them here b/c it makes it easier for clients to parse it.

But something feels off about all this candid being defined in this file, like we are violating a separation of concerns and creating unnecessary coupling.  It seems like it ought to be somewhere else, but still accessible by clients.